### ISSUE TITLE
Make users roles more centered to their location

### ISSUE THREAD

**Issue by PatriciaMoller** [SmartZoos/Toolset#174]
Jens and I have proposed a change to the user roles, as written [here](https://docs.google.com/document/d/1OAfD5CVHCLEHrwdKF8jP39Xa2uV7-Pw0Hk-9i38nYy8/).

Non-logged-in players will only be able to play Activities. Will not be able to save there play or claim rewards
Logged-in players will have their play progress saved and will be able to claim rewards. Their names will be seen 

The main takeaway is the Creator and Zoo Admin roles

The Creator will be given the rights to create and edit Activities and Act. Items. They will ONLY be able to edit what what created by the same account. This role is given to a registered user by the Zoo Admin, this to have some kind of oversight to who the access is given to. OR if this was automated process, and the Admin regularly checks the users/what they created to clean out spam. In order to maintain quality, this role should be used mainly/only by educators, event planners or personnel so the content created has a purpose and educative quality. 

The Zoo Admin will be able to create/edit anything within their location. Mainly, they will have the task of giving out Creator access. They would have a task to regularly check the newly added content (if creation rights are limited to educators/personnel, the amount of checkups could be rather low, as those people hopefully will only be adding content of higher quality)
Zoo Admin would also be in charge of adding new vouchers for creators to apply to their Activities.

This is to give the Zoos/Parks themselves more control of their own location, without having to contact any outside admins for tasks only related to them.

**Comment by PatriciaMoller** [comment]
IF Creator become a role to be used, some changes to the Dashboard to promote and make easier access to creation tools could be considered

**Comment by pjotrsavitski** [comment]
Well, those changes could be beneficial for the current situation. Users with roles of "Zoo Member", "Zoo Administrator" and "Administrator" might benefit from those.

Currently, all users are allowed to create Activities and Activity Items (for some time, at least).

**Comment by pjotrsavitski** [comment]
All those roles have that permission, thus these would all be affected by any changes. The checks are based on certain permissions, not roles as such (although those are just details).

I suppose that Zoo people would have to deal with the content, yes. We should have the policy page soon enough, will try to discuss that at the next meeting (you could mention it, just in case I forget).

**Comment by kaimikael** [comment]
Another thing:
How is communication set up between a creator and an administrator?  Currently I think we assume some kind of offline communication where a creator asks an administrator to be granted privileges, but could that be streamlined by automatically sending a message to an administrator when a new creator logs in?  Hm.  Perhaps most users only intend to play the game and not create quizzes, so some kind of button to press, ”Please make me a creator”?

**Comment by pjotrsavitski** [comment]
Yes, that is correct. Roles are currently managed by the Administrators. Those are assigned or removed manually based on some external principle. I would assume that people from the Zoos have knowledge of who their creators should be.

It could be possible to allow any user to request a certain role (creator for one of the zoos). But it should be sent to some admin then, the admin would review that and then either assign the role or reject the request.

There are a few questions about the implementation that need to be answered before implementing anything.

* Who get the request? All the administrators and the one who responds first wins?
* How would the administrator know if the user should be assigned that role or not (it could end up by being a bit quicker way of assigning roles to the trusted users; and that does not seem to be worth the effort)?
* Should the user be able to make more requests if the previous one is rejected?
* How to prevent users from abusing the functionality (one could create a lot of different account just to spam the admins)?
* Where should that button be put? Somewhere under the profile? Maybe even profile edition page?

**Comment by kaimikael** [comment]
Yeah, this is probably not a very high priority, but it got me thinking about possible workflows at the zoos and possibly everybody else has already figured out how it should work, but just to sort my mind :-)
So, the administrator(s) need(s) to determine who will be allowed to create activities.  Some ways to handle this are: 

1. The administrator decides who the creator should be beforehand, creates an account and hands over the account to the person, via email or some other message.
2. The administrator tells a person to register and either checks for their appearance or requests a message that registration has been performed.
3. A person registers, requests creator privileges and gets “promoted”.  I e, they may spend some time just playing and then decide they want to be in on creating activities.
I suspect that in the zoo situation, alternatives 1 and 2 will be the prevalent ones and 2 is the one we currently assume, whereas 3 is what I suggested.  Hm, maybe ask our zoo friends if alternative 3 is expected at all.

**Comment by PatriciaMoller** [comment]
I think a Contact form could work well.
This could be placed in "SmartZoos for education/schools" page.

Applicant can be asked to supply Email, Name, Organization and choose Closest location. And state if the have an account already or not.
Based on which location option is chosen, that location's admin gets notified. Admin then creates account with a generated password, or promotes the already existing account, and sends that back to applicant.

It would need less co-ordination if an application only got sent to one place, rather than several people in different countries. Though I guess double creating an account wouldn't be possible, it still be redundant if an admin started going through application e-mails and noticing they were all done.
There would be less chance of mixing up locations during registration as well, as you only have one to keep track of.

**Comment by kaimikael** [comment]
That sounds like a good workflow.

**Comment by PatriciaMoller** [comment]
Would it be possible to have e-mail addresses on the SmartZoos domain (eg, skansen.smartzoos.eu, etc) for all the partners?

**Comment by kaimikael** [comment]
I believe @orkit has been testing something like that.  Ryan?

**Comment by pjotrsavitski** [comment]
I will go through it in details a bit later, but a quick comment about email addresses:

Yes, we can create as many as we need, if that is required.

**Comment by PatriciaMoller** [comment]
Alternatively, an email-address hosted by the respective partner. I guess it depends on how they want to view the use of the platfrom. Is it the partner just using the platform, then a partner[at].smartzoos.eu". Or is the platform a part of the partners activities, then "smartzoos[at]partner.xx" would reflect that better. I guess it would be up the each partner to decide how they want to their address to look, but at least present the option

**Comment by PatriciaMoller** [comment]
Another alternative I just discussed with @kaimikael: Not having forms on the page, and instead have the contact info to the relevant admin where SmartZoos is featured/promoted on the partners' sites. In Skansen's case, eg. on the pages related to booking class visits and self-guiding info. 

I think this should be raised in the next face-to-face meeting, since it has everything to do with how using the platform will be.
